app/src/components/misc/SignInViaEmailLink/SignInViaEmailLink.js

```diff
@@ -57,17 +64,79 @@ const SignInViaEmailLink = () => {
     return <SpinnerColumn />;
   }, [user.email, userEmailfromLocalStorage, logOutUser]);
 
+  const updateUserFullName = useCallback(async () => {
+    if (user.details?.profile?.displayName === "User" && appOnboardingDetails.fullName) {
+      const newName = appOnboardingDetails.fullName;
+      return new Promise((resolve, reject) => {
+        updateValueAsPromise(["users", user.details?.profile?.uid, "profile"], { displayName: newName })
+          .then(() => {
+            resolve({ success: true });
+          })
+          .catch((e) => {
+            reject(e);
+          });
+      });
+    }
+    return Promise.resolve({ success: true });
+  }, [user.details?.profile?.displayName, appOnboardingDetails.fullName, user.details?.profile?.uid]);
+
+  const updateUserPersona = useCallback(async () => {
+    if (appOnboardingDetails.persona) {
+      return new Promise((resolve, reject) => {
+        setUserPersona({ persona: appOnboardingDetails.persona })
+          .then(() => {
+            resolve({ success: true });
+          })
+          .catch((e) => {
+            reject(e);
+          });
+      });
+    }
+    return Promise.resolve({ success: true });
+  }, [appOnboardingDetails.persona, setUserPersona]);
+
   useEffect(() => {
     if (user.loggedIn && isEmailLoginLinkDone) {
       const name = user?.displayName?.split(" ")[0];
       let message = isLogin ? "Welcome back to Requestly!" : "Welcome to Requestly!";
       if (name) {
         message = isLogin ? `Welcome back ${name}!` : `Welcome to Requestly ${name}!`;
       }
-      toast.success(message);
-      redirectToRules(navigate);
+      //TODO: ONLY DO THIS FOR USERS WHO ARE ASSIGNED THE EXPERIMENT
+      Promise.all([
+        updateUserPersona(),
+        updateUserFullName(),
+        updateUserInFirebaseAuthUser({
+          displayName: appOnboardingDetails.fullName || user.details?.profile?.displayName,
+        }),
+      ])
+        .then(() => {
+          //DO NOTHING
```

Comment:
add a logger log, can be useful if this breaks in prod and we have to debug through sentry sessions